### PR TITLE
Replace RestoreSources with #i

### PR DESCRIPTION
**Pull request by jonsequitur**
This replaces `#r "nuget:RestoreSources=/path/to/nuget/source"` with `#i "nuget:/path/to/nuget/source"` for the C# kernel. It still supports both URLs and directory paths.

The goal is to make this gesture work eventually in both C# script and FSI but we can pre-polyfill this for now in .NET Interactive's middleware to try it out and work out the details.

### CODE REVIEW

File: Microsoft.DotNet.Interactive.CSharp/CSharpKernelExtensions.cs [hunk 1]
```diff
@@ -84,8 +59,51 @@ public static CSharpKernel UseNugetDirective(this CSharpKernel kernel)
...
+
+        private static Command i(PackageRestoreContext restoreContext)
+        {
+            var iDirective = new Command("#i")
+            {
+                new Argument<string>("source")
+            };
+            iDirective.Handler = CommandHandler.Create<string, KernelInvocationContext>((source, context) =>
+            {
+                restoreContext.AddRestoreSource(source.Replace("nuget:", ""));
```
Comment (colombod) [inline Microsoft.DotNet.Interactive.CSharp/CSharpKernelExtensions.cs]: I think here we call PackageRestoreContext a NuGetPacakgeRestoreContext. if we want to be pluggable we should rename this type to express that we are talking about Nuget sources and NuGet packages.

File: Microsoft.DotNet.Interactive/Events/DisplayedValue.cs [hunk 2]
```diff
@@ -2,18 +2,17 @@
...
 {
-    public class DisplayedValue 
+    public class DisplayedValue
     {
         private readonly string _displayId;
         private readonly string _mimeType;
+        private readonly KernelInvocationContext _context;
 
-        public DisplayedValue(string displayId, string mimeType)
+        public DisplayedValue(string displayId, string mimeType, KernelInvocationContext context)
```
Comment (colombod) [inline Microsoft.DotNet.Interactive/Events/DisplayedValue.cs]: how will this impact the serialization of events?

Comment (jonsequitur) [reply Microsoft.DotNet.Interactive/Events/DisplayedValue.cs]: This type is not serialized.

File: Microsoft.DotNet.Interactive.CSharp/CSharpKernelExtensions.cs (hunk 1 above)
Comment (jonsequitur) [reply Microsoft.DotNet.Interactive.CSharp/CSharpKernelExtensions.cs]: I don't think we know what generalization looks like until we try to build a second implementation so it's fine for "package" to mean "NuGet package" for now. Much of the code in `PackageRestoreContext` would be the same for other package managers.

**Pull request merged**